core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -82,27 +82,28 @@
 
     @Override
     protected void completeRequest(BlockReadRequestContext context) throws Exception {
-      BlockReader reader = context.getBlockReader();
-      if (reader != null) {
-        try {
+      BlockReader reader = null;
+      try {
+        reader = context.getBlockReader();
+        if (reader != null) {
           reader.close();
-        } catch (Exception e) {
-          LOG.warn("Failed to close block reader for block {} with error {}.",
-              context.getRequest().getId(), e.getMessage());
         }
-      }
-      if (!mWorker.unlockBlock(context.getRequest().getSessionId(), context.getRequest().getId())) {
-        if (reader != null) {
-          mWorker.closeUfsBlock(context.getRequest().getSessionId(), context.getRequest().getId());
-          context.setBlockReader(null);
+      } finally {
+        if (!mWorker.unlockBlock(context.getRequest().getSessionId(),
+            context.getRequest().getId())) {
+          if (reader != null) {
+            mWorker.closeUfsBlock(context.getRequest().getSessionId(),
+                context.getRequest().getId());
+            context.setBlockReader(null);
+          }
```

Comment:
Is there a functional change here, it seems like we lose the error message of failing to close the block reader?

Reply:
good catch

Reply:
updated